aws/resource_aws_serverlessrepository_stack.go

```diff
@@ -0,0 +1,321 @@
+	}
+	if v, ok := d.GetOk("semantic_version"); ok {
+		version := v.(string)
+		changeSetRequest.SemanticVersion = aws.String(version)
+	}
+	if v, ok := d.GetOk("parameters"); ok {
+		changeSetRequest.ParameterOverrides = expandServerlessRepositoryChangeSetParameters(v.(map[string]interface{}))
+	}
+
+	log.Printf("[DEBUG] Creating Serverless Application Repository change set: %s", changeSetRequest)
+	changeSetResponse, err := serverlessConn.CreateCloudFormationChangeSet(&changeSetRequest)
+	if err != nil {
+		return fmt.Errorf("error creating Serverless Application Repository change set (%s): %w", stackName, err)
+	}
+
+	d.SetId(aws.StringValue(changeSetResponse.StackId))
+
+	_, err = cfwaiter.ChangeSetCreated(cfConn, d.Id(), aws.StringValue(changeSetResponse.ChangeSetId))
+	if err != nil {
+		return fmt.Errorf("error waiting for Serverless Application Repository change set (%s) creation: %w", stackName, err)
+	}
+
+	log.Printf("[INFO] Serverless Application Repository change set (%s) created", d.Id())
+
+	requestToken := resource.UniqueId()
+	executeRequest := cloudformation.ExecuteChangeSetInput{
+		ChangeSetName:      changeSetResponse.ChangeSetId,
+		ClientRequestToken: aws.String(requestToken),
+	}
+	log.Printf("[DEBUG] Executing Serverless Application Repository change set: %s", executeRequest)
+	_, err = cfConn.ExecuteChangeSet(&executeRequest)
+	if err != nil {
+		return fmt.Errorf("Executing Serverless Application Repository change set failed: %w", err)
+	}
+
+	_, err = cfwaiter.StackCreated(cfConn, d.Id(), requestToken, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return fmt.Errorf("error waiting for Serverless Application Repository CloudFormation Stack (%s) creation: %w", d.Id(), err)
+	}
+
+	log.Printf("[INFO] Serverless Application Repository CloudFormation Stack (%s) created", d.Id())
+
+	return resourceAwsServerlessRepositoryStackRead(d, meta)
+}
+
+func resourceAwsServerlessRepositoryStackRead(d *schema.ResourceData, meta interface{}) error {
+	serverlessConn := meta.(*AWSClient).serverlessapplicationrepositoryconn
+	cfConn := meta.(*AWSClient).cfconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	getApplicationInput := &serverlessrepository.GetApplicationInput{
+		ApplicationId: aws.String(d.Get("application_id").(string)),
+	}
+
+	_, ok := d.GetOk("semantic_version")
+	if !ok {
+		getApplicationOutput, err := serverlessConn.GetApplication(getApplicationInput)
+		if err != nil {
+			return err
+		}
+		d.Set("semantic_version", getApplicationOutput.Version.SemanticVersion)
+	}
+
+	input := &cloudformation.DescribeStacksInput{
+		StackName: aws.String(d.Id()),
+	}
+	resp, err := cfConn.DescribeStacks(input)
+	if tfawserr.ErrCodeEquals(err, "ValidationError") {
+		log.Printf("[WARN] CloudFormation stack (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return err
```

Comment:
To help operators and code maintainers with context around error messages:

```suggestion
		return fmt.Errorf("error describing CloudFormation Stack (%s): %w", d.Id(), err)
```

See also: https://github.com/hashicorp/terraform-provider-aws/issues/15892